kapitan/initialiser.py

```diff
@@ -20,27 +20,150 @@
 import os
 import sys
 import shutil
+import yaml
+
 from distutils.dir_util import copy_tree
+from kapitan.utils import list_all_paths
 
 logger = logging.getLogger(__name__)
 
-def initialise_skeleton(directory):
-    """ Initialises a directory with a recommended skeleton structure
-    Args:
-        directory (string): path which to initialise, directory is assumed to exist
-    """
-
-    current_pwd = os.path.dirname(__file__)
-    templates_directory = os.path.join(current_pwd, 'inputs', 'templates')
-
-    copy_tree(templates_directory, directory)
-
-    logger.info("Populated {} with:".format(directory))
-    for dirName, subdirList, fileList in os.walk(directory):
-        logger.info('{}'.format(dirName))
-        for fname in fileList:
-            logger.info('\t {}'.format(fname))
-        # Remove the first entry in the list of sub-directories
-        # if there are any sub-directories present
-        if len(subdirList) > 0:
-            del subdirList[0]
+class Initialiser(object):
+    def __init__(self,directory,targets,compile_inputs):
+        """ Initialises a directory with a recommended skeleton structure
+        Args:
+            directory (string): path which to initialise, directory is assumed to exist
+        """
+        self.target_template_name = 'my_target.yml'
+        self.component_template_name = 'my_component.yml'
+        self.target_dir = "inventory/targets"
```

Comment:
@harsh-98 wouldn't this cause issue with windows system?

Reply:
Python has a hack, https://docs.python.org/3/library/os.html#os.altsep. It internally, changes forwardslash to backward slash when used in windows. Ref: [here](https://medium.com/@ageitgey/python-3-quick-tip-the-easy-way-to-deal-with-file-paths-on-windows-mac-and-linux-11a072b58d5f).